### PR TITLE
RDS backups to S3

### PR DESCRIPTION
**Pull request by Stretch96**
* Conditonally create a scheduled Fargate task to backup RDS databases
  to S3
* Fargate was chosen, so that it doesn't rely on the ECS infrastructure
  to run - meaning we can have RDS backups running even if an ECS
  Infrastructure hasn't been launched
* A CodeBuild project is created, which clones, builds and pushes the
  dalmatian-sql-backup image to ECR, ready to be used for the backup
  scheduled task. The Dockerfile for this is stored at
  https://github.com/dxw/dalmatian-sql-backup. This CodeBuild project is
  triggered every 24 hours to keep up with any updates.
* The backup task will backup all databases (except system databases),
  and store the SQL dumps into an S3 bucket, which can be configured
  with the retention time in days.

### CODE REVIEW

**Comment by rjw1** [comment]
Could you add some why to this commit message. especially the decison about using fargate.

**Comment by Stretch96** [comment]
> Could you add some why to this commit message. especially the decison about using fargate.

Yup will do - Just finishing off the remaining puzzle pieces and I'll add a full commit message (I should've put WIP in the PR 😬 ...)

File: container-definitions/app.json.tpl [hunk 1]
```diff
@@ -30,10 +33,10 @@
     ],
     %{ endif }
     %{ if environment != "[]" }
-    "environment": "${environment},
```
Comment (rjw1) [inline container-definitions/app.json.tpl:33]: im slighty confused by this change.

File: ecs-entrypoints/rds-s3-backups-postgres.txt.tpl [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+&& mkdir -p /tmp/sqlbackups/${rds_identifier} \
+&& PGPASSWORD=\"$$DB_PASSWORD\" \
+psql \
+  -U $$DB_USERNAME \
+  -h $$DB_HOST \
+  -t \
+  -c 'SELECT datname FROM pg_database WHERE NOT datistemplate' \
+| while read DB_NAME; \
+do \
+  if [[ -n \"$$DB_NAME\" && \"$$DB_NAME\" != \"rdsadmin\" ]]; \
```
Comment (rjw1) [inline ecs-entrypoints/rds-s3-backups-postgres.txt.tpl:11]: while not important for this PR. I wonder if this shoudl move to using the grep like the mysql one.

**Review by rjw1 — APPROVED** [review]
Looks goos. i have a suggestion and a befuddlement but they shouldnt stop this being merged.

File: container-definitions/app.json.tpl (hunk 1 above)
Comment (Stretch96) [reply container-definitions/app.json.tpl:33]: That was a fix from the previous PR ... It shouldn't have the `"`

Comment (Stretch96) [reply container-definitions/app.json.tpl:33]: (`environment` hasn't been used until this PR)

File: ecs-entrypoints/rds-s3-backups-postgres.txt.tpl (hunk 2 above)
Comment (Stretch96) [reply ecs-entrypoints/rds-s3-backups-postgres.txt.tpl:11]: Okie yep can do - I'll do that when we move these to scripts within the Docker image

**Pull request merged**